.github/workflows/checks.yml

```diff
@@ -15,6 +15,12 @@ jobs:
         ]
 
     runs-on: ${{ matrix.os }}
+    env:
+      STARKNET_JVM_INTEGRATION_TEST_SOURCE: "env"
+      STARKNET_JVM_SKIP_INTEGRATION_GETTERS_TEST: "false"
```

Comment:
It is.
- https://github.com/software-mansion/starknet-jvm/blob/09d20093dc7be422db0abe891c6e64c83314e7b1/lib/src/test/kotlin/integration/utils/ConfigUtils.kt#L69C1-L70C1
Used to determine whether to get secrets from file or env.
- https://github.com/software-mansion/starknet-jvm/blob/09d20093dc7be422db0abe891c6e64c83314e7b1/lib/src/test/kotlin/integration/getters/GettersTest.kt#L47C10-L47C10
Checking whether we want to run a test on CI. This is useful because node could be down, not updated, or we do not want run some gas-heavy tests on CI.

Reply:
1st link links to `STARKNET_JVM_INTEGRATION_TEST_SOURCE`
2nd link links to `assumeFalse(ConfigUtils.isTestSetSkipped(testName), "$testName is skipped in the current environment.")`

I assume you wanted to link this https://github.com/software-mansion/starknet-jvm/blob/09d20093dc7be422db0abe891c6e64c83314e7b1/lib/src/test/kotlin/integration/utils/ConfigUtils.kt#L38

I think you can make it a lot simpler and just do "CI_SKIP_TEST", and that's it. no need to get `$screamingSnakeName` there

Reply:
1st links links to where it's used from what I see?
2nd link:
> I think you can make it a lot simpler and just do "CI_SKIP_TEST", and that's it. no need to get $screamingSnakeName there
This what I did at first, but I assume we are planning to add the integration tests for functions that do change the state and skip those on CI. So I thought we could make it so we have a variable for every test file.

Reply:
> 1st links links to where it's used from what I see?

My question was about `STARKNET_JVM_SKIP_INTEGRATION_GETTERS_TEST`, and it's not used where 1st link points to

> This what I did at first, but I assume we are planning to add the integration tests for functions that do change the state and skip those on CI. So I thought we could make it so we have a variable for every test file.

Then you will have as many variables as test files, right? All doing basically the same thing - stopping tests from execution (presumably on CI). It seems to me, that one variable should be enough to achieve exactly this

Reply:
Well, I'm intending to run GettersTest on CI and run tests that require gas only locally. Can't achieve that with one variable.

Reply:
why?

Reply:
Ok, do I understand correctly that your idea is that for every test file that is not intended to be ran on CI we will have a check in `@BeforeAll`, and for those which are intended to run wherever we won't .

Reply:
Basically - yes. 

Although, I don't think you **_need_** to have a `BeforeAll`, we could, for example, flip the scenario and have a `RUN_INTEGRATION_TEST` variable that, when it's set, triggers those tests (by default - no variable, so tests won't run), and have a check in gradle script, something like this: https://www.baeldung.com/gradle-skip-tests#using-the-gradle-build-script. That would probably make some sense, as those tests won't be triggered very often, due to the cost etc

Reply:
I just think that modifying this: 
```yml
STARKNET_JVM_SKIP_INTEGRATION_GETTERS_TEST: "false"
STARKNET_JVM_SKIP_INTEGRATION_ACCOUNT_TEST: "true"
STARKNET_JVM_SKIP_INTEGRATION_PROVIDER_TEST: "true"
```
is a bit simplier than modifying lines of code in every test. 

If I want to disable getters test both on integration and locally (because node is down etc.), I don't need to go to that specific file and look where do I have a check and modify any code - I just change a variable.

Reply:
I think you misunderstood what I wrote.

My proposition was to run integration tests more or less like so:

```
./gradlew lib:test -Pintegration
```

in this scenario, 

```
./gradlew lib:test
```

does not run integration tests.

Reply:
> My proposition was to run integration tests more or less like so:
> `./gradlew lib:test -Pintegration`
> in this scenario,
> `./gradlew lib:test`
> does not run integration tests.

Ok, but I'm not sure how would that be better. It just seems to me that it is more convenient to have everything checked after running `./gradlew lib:test`, unless something is disabled explicitly. Specifically GettersTest are fast to run, so don't know why not include them in lib:test by default.

> edit: OR using env variable (this was not covered in the link I provided, but still I think may be a good idea) - just checking the variable, and if it not set (or false) - don't run integration tests; othwerwise do. No need to modify any code.
Sorry, I cannot understand how can you achieve specifying whether test should be skipped with just one variable without having different code for every test. 😅

Reply:
> Ok, but I'm not sure how would that be better. It just seems to me that it is more convenient to have everything checked after running ./gradlew lib:test, unless something is disabled explicitly. Specifically GettersTest are fast to run, so don't know why not include them in lib:test by default.

but this is very similar approach, the only thing is that you don't have an env variable, but you pass an argument directly to the test command. 

I know they are fast to run, but they are also prone to errors - what if rpc node is down? what if it has wrong version? what if someone does not have a rpc node, but wants to contribute with some simple PR? I jsut don't think we should run such tests by default, only on demand.

> Sorry, I cannot understand how can you achieve specifying whether test should be skipped with just one variable without having different code for every test. 😅

depends on approach, let's say for the sake of an argument that the approach does not change - everything is the same as in your PR. Then you just replace `STARKNET_JVM_SKIP_INTEGRATION_GETTERS_TEST`, `STARKNET_JVM_SKIP_INTEGRATION_ACCOUNT_TEST`, `STARKNET_JVM_SKIP_INTEGRATION_PROVIDER_TEST` variable checking in your code to one variable `SKIP_CI_TEST`, and that's it.